modules/utxo-lib/src/bitgo/Unspent.ts

```diff
@@ -110,6 +124,66 @@ export function addToTransactionBuilder<TNumber extends number | bigint>(
   txb.addInput(txid, vout, sequence, script, value);
 }
 
+export function addToPsbt(
+  psbt: UtxoPsbt<UtxoTransaction<bigint>>,
+  u: WalletUnspent<bigint>,
+  rootSigner: WalletUnspentSigner<RootWalletKeys>,
+  network: Network
+): void {
+  const { txid, vout, script, value } = toPrevOutput(u, network);
+  const { walletKeys, signerIndex, cosignerIndex } = rootSigner.deriveForChainAndIndex(u.chain, u.index);
+  const scriptType = scriptTypeForChain(u.chain);
+  psbt.addInput({
+    hash: txid,
+    index: vout,
+  });
+  const inputIndex = psbt.inputCount - 1;
+  if (isSegwit(u.chain)) {
+    psbt.updateInput(inputIndex, {
+      witnessUtxo: {
+        script,
+        value,
+      },
+    });
+  } else {
+    if (!isNonWitnessUnspent(u)) {
+      throw new Error('Error, require previous tx to add to PSBT');
+    }
+    psbt.updateInput(inputIndex, { nonWitnessUtxo: u.prevTx });
+  }
+
+  if (scriptType === 'p2tr') {
+    const { controlBlock, witnessScript, leafVersion, leafHash } = createSpendScriptP2tr(walletKeys.publicKeys, [
+      walletKeys.triple[signerIndex].publicKey,
+      walletKeys.triple[cosignerIndex].publicKey,
+    ]);
+    psbt.updateInput(inputIndex, {
```

Comment:
In the case of a segwit unspent, we're calling `updateInput` twice. Once on line 142 and then again either here or in the `else` block for non-p2tr. Is there a reason we have to call it twice? Or can it be in one call? Where we'd check for either non-segwit, segwit p2tr, or segwit non-p2tr and have one updateInput call for each.

Reply:
Could do it in one call, but it's more appropriate to do it in separate calls, because of the logic flow.

"Any segwit gets a witnessUtxo, taproot gets taproot fields, segwitv0 gets witnessScript"

`updateInput` is intended to be called many times, often by separate updaters (eg. one might have access to a full node to fetch UTXOs, and another might have access to pubkeys to generate scripts).